lib/Cake/Utility/ObjectCollection.php

```diff
@@ -126,7 +126,7 @@ public function trigger($callback, $params = array(), $options = array()) {
 		}
 		$result = null;
 		foreach ($list as $name) {
-			$result = call_user_func_array(array($this->_loaded[$name], $callback), array_filter(compact('subject')) + $params);
+			$result = call_user_func_array(array($this->_loaded[$name], $callback), array_values(array_filter(compact('subject')) + $params));
```

Comment:
```suggestion
			$result = call_user_func_array(array($this->_loaded[$name], $callback), array_filter(array($subject)) + $params);
```
Let's make this simpler.

Reply:
@othercorey this breaks the test suite for me (right now tested this on 7.4):
```
ArgumentCountError : Too few arguments to function ContainableBehavior::beforeFind(), 1 passed in /app/Vendor/cakephp/cakephp/lib/Cake/Utility/ObjectCollection.php on line 129 and exactly 2 expected
 /app/Vendor/cakephp/cakephp/lib/Cake/Model/Behavior/ContainableBehavior.php:93
 /app/Vendor/cakephp/cakephp/lib/Cake/Utility/ObjectCollection.php:129
 /app/Vendor/cakephp/cakephp/lib/Cake/Event/CakeEventManager.php:244
 /app/Vendor/cakephp/cakephp/lib/Cake/Model/Model.php:3110
 /app/Vendor/cakephp/cakephp/lib/Cake/Model/Model.php:3020
 /app/Model/AppModel.php:24
 /app/Vendor/cakephp/cakephp/lib/Cake/Model/Datasource/DboSource.php:636
 /app/Vendor/cakephp/cakephp/lib/Cake/Model/Model.php:834
 /app/Test/…

```

Reply:
It should be the same except without the $subject key. What is missing?

Reply:
PS: was now able to test in PHP8, same issue/stacktrace.